crates/chain/src/chain_data.rs

```diff
@@ -186,6 +189,9 @@ impl AnchorFromBlockPosition for ConfirmationHeightAnchor {
 /// An [`Anchor`] implementation that also records the exact confirmation time and height of the
 /// transaction.
 ///
+/// Note that the confirmation block and the anchor block can be different here. If you
+/// know they're always going to be the same, you can use the [`BlockId`] anchor instead.
```

Comment:
This advice doesn't make sense I guess since you would lose the `confirmation_time` info (unless we had a way of storing `confirmation_time` in `LocalChain`/`CheckPoint` -- we could turn this into an issue if we feel enthusiastic about the idea but it might be too early).

Reply:
Ah, I didn't notice this. Thanks! I wonder if it would make sense to have some comment explaining the differences between the various anchor implementations, and if so, where it would make sense to put it. I feel like it's something important to have, but that doesn't really belong to the `Anchor` trait.

Reply:
Yes I think so. It's a good idea to explain why each anchor exists on the implementation. It's mostly reflective of what we get from blockchains.